sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/firestore/FirestoreStatefulComponentFactory.java

```diff
@@ -86,7 +90,11 @@ FirestoreStub getFirestoreStub(PipelineOptions options) {
 
       FirestoreOptions firestoreOptions = options.as(FirestoreOptions.class);
       String emulatorHostPort = firestoreOptions.getEmulatorHost();
+      if (emulatorHostPort == null) {
+        emulatorHostPort = System.getenv(FIRESTORE_EMULATOR_HOST_ENV_VARIABLE);
+      }
       if (emulatorHostPort != null) {
+        firestoreOptions.setEmulatorHost(emulatorHostPort);
```

Comment:
Injecting the value back into the value given to us does not make sense -- we should be treating the options argument as immutable.

Reply:
Removed the update back in FirestoreOptions.